### PR TITLE
chore: use a single Jenkinsfile for both ci.jenkins.io and infra.ci.jenkins.io

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR removes the instruction added to test https://github.com/jenkins-infra/pipeline-library/pull/784, and use a single Jenkinsfile for both ci.jenkins.io and infra.ci.jenkins.io.

Related:
- https://github.com/jenkins-infra/kubernetes-management/pull/4969

Ref:
- https://github.com/jenkins-infra/account-app/pull/351#discussion_r1485875113

### CODE REVIEW

**Review by dduportal — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -1,8 +1,6 @@
-@Library('pipeline-library@pull/784/head') _
-
 pipeline {
     agent {
-        label 'jdk17'
+        label 'jdk17 || linux-arm64'
```
Comment (dduportal) [inline Jenkinsfile:3]: Build fails on infra.ci.jenkins.io with the following error:

```
08:26:45  * What went wrong:

08:26:45  Could not determine the dependencies of task ':javadoc'.

08:26:45  > Failed to calculate the value of task ':compileJava' property 'javaCompiler'.

08:26:45     > Cannot find a Java installation on your machine matching this tasks requirements: {languageVersion=17, vendor=any, implementation=vendor-specific} for LINUX on aarch64.

08:26:45        > No locally installed toolchains match and toolchain download repositories have not been configured.
```

=> looks like we need to amend this pipeline with a `tool` directive

**Review by dduportal — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (lemeurherve) [reply Jenkinsfile:3]: Note: add a comment about arm, already done in the Dockerfile via platforms

Comment (lemeurherve) [reply Jenkinsfile:3]: And an issue to build with arm on ci.jenkins.io too like the platform used in prod.

Comment (lemeurherve) [inline Jenkinsfile:3]:
```suggestion
        // infra.ci build on arm64 in the Dockerfile, as it's used in production
        label 'jdk17 || linux-arm64'
```

**Comment by dduportal** [comment]
The build passes: checks fails at the "Test" stage with the following error:

```
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)
```

I wonder if this pod template (on infra.ci) has enough resources (to compare with the `jdk17` on ci.jenkins.io).

**Comment by lemeurherve** [comment]
> I wonder if this pod template (on infra.ci) has enough resources (to compare with the `jdk17` on ci.jenkins.io).

It doesn't seem so...

On ci.jenkins.io:
```
00:00:19.731      resources:
00:00:19.731        limits:
00:00:19.731          memory: "8G"
00:00:19.731          cpu: "4"
00:00:19.731        requests:
00:00:19.731          memory: "8G"
00:00:19.731          cpu: "4"
```

On infra.ci.jenkins.io:
```
00:00:14.482      resources:
00:00:14.482        limits:
00:00:14.482          memory: "1024Mi"
00:00:14.482          cpu: "500m"
00:00:14.482        requests:
00:00:14.482          memory: "512Mi"
00:00:14.482          cpu: "500m"
```
😅 

Adding a custom pod template for this job.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,8 +1,7 @@
-@Library('pipeline-library@pull/784/head') _
-
 pipeline {
     agent {
-        label 'jdk17'
+        // infra.ci build on arm64 in the Dockerfile, as it's used in production
+        label 'jdk17 || linux-large-arm64'
```
Comment (lemeurherve) [inline Jenkinsfile:4]:
```suggestion
        label 'jdk17 || linux-arm64-docker'
```
Docker is needed, using a (large) VM in infra.ci.jenkins.io

**Comment by dduportal** [comment]
Build step is failing on infra.ci.jenkins.io with the following:

```
13:05:19  FAILURE: Build failed with an exception.
13:05:19  
13:05:19  * What went wrong:
13:05:19  Could not determine the dependencies of task ':javadoc'.
13:05:19  > Failed to calculate the value of task ':compileJava' property 'javaCompiler'.
13:05:19     > Cannot find a Java installation on your machine matching this tasks requirements: {languageVersion=17, vendor=any, implementation=vendor-specific} for LINUX on aarch64.
13:05:19        > No locally installed toolchains match and toolchain download repositories have not been configured.
13:05:19  
13:05:19  * Try:
13:05:19  > Learn more about toolchain auto-detection at https://docs.gradle.org/8.6/userguide/toolchains.html#sec:auto_detection.
13:05:19  > Learn more about toolchain repositories at https://docs.gradle.org/8.6/userguide/toolchains.html#sub:download_repositories.
13:05:19  > Run with --stacktrace option to get the stack trace.
13:05:19  > Run with --info or --debug option to get more log output.
13:05:19  > Run with --scan to get full insights.
13:05:19  > Get more help at https://help.gradle.org/
13:05:19 
13:05:19  Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0
13:05:19
13:05:19  You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
```


=> triggered a replay with `java --version` and it shows:

```
20:28:24  + java --version
20:28:26  openjdk 11.0.23 2024-04-16
20:28:26  OpenJDK Runtime Environment Temurin-11.0.23+9 (build 11.0.23+9)
20:28:26  OpenJDK 64-Bit Server VM Temurin-11.0.23+9 (build 11.0.23+9, mixed mode)
```


while the project expect JDK17.

=> It means that the Linux arm64 VM template is misconfigured on infra.ci:

- Either it specifies a default JAVA_HOME to the JDK11
- Or the packer template image defines JDK11 as default JDK

**Comment by smerle33** [comment]
switched back to amd64 for selenium : 
```
Caused by: org.openqa.selenium.WebDriverException: Driver server process died prematurely.
```

**Review by dduportal — APPROVED** [review]

**Pull request merged**